### ISSUE TITLE
data race in simulator/view_manager

### ISSUE THREAD

**Issue by toomasp** [vmware/govmomi#1620]
Found while goroutines were using the same ViewManager

```
WARNING: DATA RACE
Write at 0x00c00050c0a0 by goroutine 94:
  github.com/vmware/govmomi/simulator.(*ViewManager).CreateContainerView()
      C:/Users/toomasp/IdeaProjects/vcenter_scanner.go/vendor/github.com/vmware/govmomi/simulator/view_manager.go:122 +0x737

Previous read at 0x00c00050c0a0 by goroutine 25:
  github.com/vmware/govmomi/simulator.(*ViewManager).Reference()
      <autogenerated>:1 +0x74
  github.com/vmware/govmomi/simulator.(*Registry).locker()
      C:/Users/toomasp/IdeaProjects/vcenter_scanner.go/vendor/github.com/vmware/govmomi/simulator/registry.go:505 +0xf2
```

**Comment by dougm** [comment]
I think I see where the problem is, if you have a test that can reproduce the race that would be most helpful.  Ideally one we can add to https://github.com/vmware/govmomi/blob/master/simulator/view_manager_test.go

**Comment by toomasp** [comment]
https://github.com/toomasp/govmomi/commit/67dc27a26070904e8d411f2c3db16814c4a7f896

**Comment by dougm** [comment]
Thank you @toomasp , can you review the fix in #1632 ?

**Comment by toomasp** [comment]
Did, appears that it doesn't trigger reliably enough ( sorry for that, but it seems that I just lucked out when testing it )

```bash
toomasp@playtech-v1914  ~/tmp/govmomi  ➦ 9b0db1c  GORACE=history_size=5 go test -race -count 1 -v ./simulator -test.run ^TestViewManager_CreateContainerView$
=== RUN   TestViewManager_CreateContainerView
--- PASS: TestViewManager_CreateContainerView (1.05s)
PASS
ok  	github.com/vmware/govmomi/simulator	2.139s
 toomasp@playtech-v1914  ~/tmp/govmomi  ➦ 9b0db1c  GORACE=history_size=5 go test -race -count 1 -v ./simulator -test.run ^TestViewManager_CreateContainerView$
=== RUN   TestViewManager_CreateContainerView
==================
WARNING: DATA RACE
Read at 0x00c000032c20 by goroutine 76:
  github.com/vmware/govmomi/simulator.(*ViewManager).Reference()
      <autogenerated>:1 +0x6d
  github.com/vmware/govmomi/simulator.(*Registry).locker()
      /home/toomasp/tmp/govmomi/simulator/registry.go:505 +0xeb
  github.com/vmware/govmomi/simulator.(*Registry).WithLock()
      /home/toomasp/tmp/govmomi/simulator/registry.go:528 +0x88
  github.com/vmware/govmomi/simulator.(*Service).call()
      /home/toomasp/tmp/govmomi/simulator/simulator.go:208 +0x5c0
  github.com/vmware/govmomi/simulator.(*Service).ServeSDK()
      /home/toomasp/tmp/govmomi/simulator/simulator.go:439 +0x16b5
  github.com/vmware/govmomi/simulator.(*Service).ServeSDK-fm()
      /home/toomasp/tmp/govmomi/simulator/simulator.go:399 +0x5f
  net/http.HandlerFunc.ServeHTTP()
      /opt/go/src/net/http/server.go:1995 +0x51
  net/http.(*ServeMux).ServeHTTP()
      /opt/go/src/net/http/server.go:2375 +0x28a
  net/http.serverHandler.ServeHTTP()
      /opt/go/src/net/http/server.go:2774 +0xce
  net/http.(*conn).serve()
      /opt/go/src/net/http/server.go:1878 +0x811

Previous write at 0x00c000032c20 by goroutine 77:
  github.com/vmware/govmomi/simulator.(*ViewManager).CreateContainerView()
      /home/toomasp/tmp/govmomi/simulator/view_manager.go:122 +0x73b
  runtime.call64()
      /opt/go/src/runtime/asm_amd64.s:520 +0x3a
  reflect.Value.Call()
      /opt/go/src/reflect/value.go:308 +0xc0
  github.com/vmware/govmomi/simulator.(*Service).call.func1()
      /home/toomasp/tmp/govmomi/simulator/simulator.go:209 +0x7b
  github.com/vmware/govmomi/simulator.(*Registry).WithLock()
      /home/toomasp/tmp/govmomi/simulator/registry.go:532 +0x51
  github.com/vmware/govmomi/simulator.(*Service).call()
      /home/toomasp/tmp/govmomi/simulator/simulator.go:208 +0x5c0
  github.com/vmware/govmomi/simulator.(*Service).ServeSDK()
      /home/toomasp/tmp/govmomi/simulator/simulator.go:439 +0x16b5
  github.com/vmware/govmomi/simulator.(*Service).ServeSDK-fm()
      /home/toomasp/tmp/govmomi/simulator/simulator.go:399 +0x5f
  net/http.HandlerFunc.ServeHTTP()
      /opt/go/src/net/http/server.go:1995 +0x51
  net/http.(*ServeMux).ServeHTTP()
      /opt/go/src/net/http/server.go:2375 +0x28a
  net/http.serverHandler.ServeHTTP()
      /opt/go/src/net/http/server.go:2774 +0xce
  net/http.(*conn).serve()
      /opt/go/src/net/http/server.go:1878 +0x811

Goroutine 76 (running) created at:
  net/http.(*Server).Serve()
      /opt/go/src/net/http/server.go:2884 +0x4c4
  github.com/vmware/govmomi/simulator/internal.(*Server).goServe.func1()
      /home/toomasp/tmp/govmomi/simulator/internal/server.go:268 +0xac

Goroutine 77 (running) created at:
  net/http.(*Server).Serve()
      /opt/go/src/net/http/server.go:2884 +0x4c4
  github.com/vmware/govmomi/simulator/internal.(*Server).goServe.func1()
      /home/toomasp/tmp/govmomi/simulator/internal/server.go:268 +0xac
==================
==================
WARNING: DATA RACE
Read at 0x00c000776020 by goroutine 54:
  github.com/vmware/govmomi/simulator.RemoveReference()
      /home/toomasp/tmp/govmomi/simulator/registry.go:394 +0x90
  github.com/vmware/govmomi/simulator.destroyView()
      /home/toomasp/tmp/govmomi/simulator/view_manager.go:74 +0x8a
  github.com/vmware/govmomi/simulator.(*ContainerView).DestroyView()
      /home/toomasp/tmp/govmomi/simulator/view_manager.go:144 +0x105
  runtime.call64()
      /opt/go/src/runtime/asm_amd64.s:520 +0x3a
  reflect.Value.Call()
      /opt/go/src/reflect/value.go:308 +0xc0
  github.com/vmware/govmomi/simulator.(*Service).call.func1()
      /home/toomasp/tmp/govmomi/simulator/simulator.go:209 +0x7b
  github.com/vmware/govmomi/simulator.(*Registry).WithLock()
      /home/toomasp/tmp/govmomi/simulator/registry.go:532 +0x51
  github.com/vmware/govmomi/simulator.(*Service).call()
      /home/toomasp/tmp/govmomi/simulator/simulator.go:208 +0x5c0
  github.com/vmware/govmomi/simulator.(*Service).ServeSDK()
      /home/toomasp/tmp/govmomi/simulator/simulator.go:439 +0x16b5
  github.com/vmware/govmomi/simulator.(*Service).ServeSDK-fm()
      /home/toomasp/tmp/govmomi/simulator/simulator.go:399 +0x5f
  net/http.HandlerFunc.ServeHTTP()
      /opt/go/src/net/http/server.go:1995 +0x51
  net/http.(*ServeMux).ServeHTTP()
      /opt/go/src/net/http/server.go:2375 +0x28a
  net/http.serverHandler.ServeHTTP()
      /opt/go/src/net/http/server.go:2774 +0xce
  net/http.(*conn).serve()
      /opt/go/src/net/http/server.go:1878 +0x811

Previous write at 0x00c000776020 by goroutine 78:
  runtime.slicecopy()
      /opt/go/src/runtime/slice.go:197 +0x0
  github.com/vmware/govmomi/simulator.RemoveReference()
      /home/toomasp/tmp/govmomi/simulator/registry.go:396 +0x1ef
  github.com/vmware/govmomi/simulator.destroyView()
      /home/toomasp/tmp/govmomi/simulator/view_manager.go:74 +0x8a
  github.com/vmware/govmomi/simulator.(*ContainerView).DestroyView()
      /home/toomasp/tmp/govmomi/simulator/view_manager.go:144 +0x105
  runtime.call64()
      /opt/go/src/runtime/asm_amd64.s:520 +0x3a
  reflect.Value.Call()
      /opt/go/src/reflect/value.go:308 +0xc0
  github.com/vmware/govmomi/simulator.(*Service).call.func1()
      /home/toomasp/tmp/govmomi/simulator/simulator.go:209 +0x7b
  github.com/vmware/govmomi/simulator.(*Registry).WithLock()
      /home/toomasp/tmp/govmomi/simulator/registry.go:532 +0x51
  github.com/vmware/govmomi/simulator.(*Service).call()
      /home/toomasp/tmp/govmomi/simulator/simulator.go:208 +0x5c0
  github.com/vmware/govmomi/simulator.(*Service).ServeSDK()
      /home/toomasp/tmp/govmomi/simulator/simulator.go:439 +0x16b5
  github.com/vmware/govmomi/simulator.(*Service).ServeSDK-fm()
      /home/toomasp/tmp/govmomi/simulator/simulator.go:399 +0x5f
  net/http.HandlerFunc.ServeHTTP()
      /opt/go/src/net/http/server.go:1995 +0x51
  net/http.(*ServeMux).ServeHTTP()
      /opt/go/src/net/http/server.go:2375 +0x28a
  net/http.serverHandler.ServeHTTP()
      /opt/go/src/net/http/server.go:2774 +0xce
  net/http.(*conn).serve()
      /opt/go/src/net/http/server.go:1878 +0x811

Goroutine 54 (running) created at:
  net/http.(*Server).Serve()
      /opt/go/src/net/http/server.go:2884 +0x4c4
  github.com/vmware/govmomi/simulator/internal.(*Server).goServe.func1()
      /home/toomasp/tmp/govmomi/simulator/internal/server.go:268 +0xac

Goroutine 78 (running) created at:
  net/http.(*Server).Serve()
      /opt/go/src/net/http/server.go:2884 +0x4c4
  github.com/vmware/govmomi/simulator/internal.(*Server).goServe.func1()
      /home/toomasp/tmp/govmomi/simulator/internal/server.go:268 +0xac
==================
--- FAIL: TestViewManager_CreateContainerView (1.19s)
    testing.go:809: race detected during execution of test
FAIL
FAIL	github.com/vmware/govmomi/simulator	1.237s
```
